### PR TITLE
vprof - visual profiler

### PR DESCRIPTION
**Pull request by alen12345**

### CODE REVIEW

File: recipes/vprof/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - python
+
+test:
+  imports:
+    - vprof
+    - vprof.tests
+
+about:
+  home: https://github.com/nvdv/vprof
+  license: 'BSD 2'
```
Comment (CJ-Wright) [inline recipes/vprof/meta.yaml]: BSD-2-Clause

Comment (alen12345) [reply recipes/vprof/meta.yaml]: Fixing in next commit, thank you!

File: recipes/vprof/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{% set name = "vprof" %}
+{% set developer = "nvdv" %}
```
Comment (CJ-Wright) [inline recipes/vprof/meta.yaml]: Would you mind hard coding this? https://github.com/conda-forge/staged-recipes/pull/4734#discussion_r160053925

File: recipes/vprof/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{% set version = "0.37.4" %}
+{% set sha256 = "83fc52944a00cb488e0297e90f8ff4ac1d4c8b39c30563879058a1455005ddc4" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
+  url: https://github.com/{{ developer}}/{{ name }}/archive/v{{ version }}.tar.gz
```
[the remark this answers is not in the document]
Comment (alen12345) [reply recipes/vprof/meta.yaml]: I haven't understood :( can you please explain? Do you mean the package version?

File: recipes/vprof/meta.yaml (hunk 2 above)
Comment (alen12345) [reply recipes/vprof/meta.yaml]: Yes sure! I had deleted the comment and refactored like this. Next commit will be hard coded!

File: recipes/vprof/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (CJ-Wright) [reply recipes/vprof/meta.yaml]: No @justcalamari and I are working on an auto version ticking project. This url template is a bit unusual, so I was curious if that would work with the engine of the project. (this isn't really a thing about your package, so don't worry)

**Comment by alen12345** [comment]
Please don't merge yet, we are still missing tests. I will write a final post when I am done :)

**Comment by CJ-Wright** [comment]
Just ping me when you're done.

**Comment by alen12345** [comment]
@CJ-Wright I have also fixed the tests!

**Comment by CJ-Wright** [comment]
Windows seems unhapy

**Comment by CJ-Wright** [comment]
If you want you can skip windows and try to fix it in the feedstock.
I am not very good at windows stuff so @conda-forge/staged-recipes thoughts?

File: recipes/vprof/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
+  url: https://github.com/nvdv/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py deps_install && python setup.py build_ui && python setup.py install && python setup.py test && python setup.py e2e_test
```
Comment (isuruf) [inline recipes/vprof/meta.yaml]: > python setup.py deps_install

Deps should be installed using conda

Comment (CJ-Wright) [reply recipes/vprof/meta.yaml]: Thank you for the catch!

File: recipes/vprof/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (justcalamari) [reply recipes/vprof/meta.yaml]: @CJ-Wright I'm pretty sure this will work for auto-tick

[the remark this answers is not in the document]
Comment (CJ-Wright) [reply recipes/vprof/meta.yaml]: Awesome!

**Comment by alen12345** [comment]
@CJ-Wright I see now that each test environment is for a different OS. Window is in fact not happy. I will try to fix that too. I will also move that part @isuruf !

**Comment by alen12345** [comment]
@isuruf unfortunately we cannot remove `python setup.py deps_install`, because we need to install also nodejs dependencies; I have anyhow added the dependencies of requirements.txt in conda too. 

In the meantime I will also try to fix the issue with Windows.

**Comment by alen12345** [comment]
Same here, test on Windows are not triggered and executed.

**Pull request closed without merging**